server.js

```diff
@@ -3,35 +3,122 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import netflixData from './data/netflix-titles.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
+//model for the entries in the database
+const Title = mongoose.model('Title', {
+  show_id: Number,
+  title: String,
+  director: String,
+  cast: String,
+  country: String,
+  date_added: String,
+  release_year: Number,
+  rating: String,
+  duration: String,
+  listed_in: String,
+  description: String,
+  type: String,
+})
+
+//seeding the database
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Title.deleteMany({})
+
+    netflixData.forEach((titleData) => {
+      new Title(titleData).save()
+    })
+  }
+  seedDatabase()
+}
+
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 8084
 const app = express()
 
+const listEndpoints = require('express-list-endpoints')
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
 })
 
+//route to all titles & queries
+
+app.get('/titles', async (req, res) => {
+  const { title, cast, country } = req.query
+  //i means to ignore case
+  const titleRegex = new RegExp(title, 'i')
+  const castRegex = new RegExp(cast, 'i')
+  const countryRegex = new RegExp(country, 'i')
+  //doesn't work when I use this Regex so it only searches for entire words. It works for the queries but it won't find titles anymore.
+  //   const titleRegex = new RegExp('\\b' + title + '\\b', 'i')
+  //   const castRegex = new RegExp('\\b' + cast + '\\b', 'i')
+  //   const countryRegex = new RegExp('\\b' + country + '\\b', 'i')
```

Comment:
You need to choose if you want to use `//` or `RegExp()` notation.
Here you can see how to use `/b` flag properly:
https://www.w3schools.com/Jsref/jsref_regexp_begin.asp